src/bin/bat/app.rs

```diff
@@ -338,4 +341,38 @@ impl App {
 
         Ok(styled_components)
     }
+
+    fn header_components(&self) -> Result<HeaderComponents> {
+        let matches = &self.matches;
+        let header_components = HeaderComponents({
+            let env_header_components: Option<Vec<HeaderComponent>> = env::var("BAT_HEADER_INFO")
+                .ok()
+                .map(|header_str| {
+                    header_str
+                        .split(',')
+                        .map(HeaderComponent::from_str)
+                        .collect::<Result<Vec<HeaderComponent>>>()
+                })
+                .transpose()?;
+
+            matches
+                .values_of("header-info")
+                .map(|header| {
+                    header
+                        .map(|header| header.parse::<HeaderComponent>())
+                        .filter_map(|header| header.ok())
+                        .collect::<Vec<_>>()
+                })
+                .or(env_header_components)
+                .unwrap_or_else(|| vec![HeaderComponent::Full])
```

Comment:
The default I've set to `Full` now, but I'm not sure if we should have the default as `Filename` alone, to be backward compatible?

Reply:
I actually changed this to be `Filename` since I noticed a lot of tests were failing as well.